.github/workflows/gen_pr.yaml

```diff
@@ -17,41 +17,44 @@ on:
   push:
     branches:
       - "main"
+
 jobs:
   create_pr:
     runs-on: ubuntu-18.04
     steps:
     - name: Check out repo
       uses: actions/checkout@v3
+      
     - name: Setup Node for gen/postgen scripts
       uses: actions/setup-node@v3
       with:
         node-version: '12'
+
     - name: Setup Go for proto2jsonschema script
       uses: actions/setup-go@v3
       with:
         go-version: '1.14'
+
     - name: Run the generators
       run: ./scripts/gen.sh
-    - name: Commit changelog and manifest files
+
+    - name: Check for diff
+      run: git diff --exit-code # exits with code 1 if diff is found
+
+    - name: Create pull request for changes
+      uses: googleapis/code-suggester@v2
+      if: ${{ failure() }} # run if diff exists
```

Comment:
For my own learning in GHA convention, would it make sense to inline the git diff here instead of adding the overhead of the separate step? What are we getting from the separation?